### ISSUE TITLE
Move request configuration into a common trait

### ISSUE THREAD

**Issue by sagebind** [sagebind/isahc#48]
There are two ways of configuring request execution:

1. Create a custom client using `HttpClientBuilder`, which provides inherent methods for configuring defaults that will be used for all requests executed by the client.
2. Configure an individual request using extension trait methods provided by `RequestBuilderExt`. The provided methods have the same names and documentation as the above methods.

To centralize documentation and avoid having two methods with the same in the public API, it would be helpful to centralize these methods into a common trait. Such a trait might look like this:

```rust
pub trait Configurable {
    /// Set a timeout for the maximum time allowed for a request-response cycle.
    ///
    /// If not set, no timeout will be enforced.
    fn timeout(self, timeout: Duration) -> Self;

    // more methods...
}

// Clients are configurable...
impl Configurable for HttpClientBuilder {
    // ...
}

// and so are requests.
impl Configurable for http::request::Builder {
    // ...
}
```

~This is not currently possible, because `HttpClientBuilder` is a by-value builder (that takes `self`, while `http::request::Builder` is a mutable builder (takes `&mut self`), so the methods cannot have the same signature.~

~It appears that in the 0.2 next release of http, the builders may change to also be by-value: https://github.com/hyperium/http/issues/191. If that is the case, it may be best to introduce this change at the same time as when we upgrade our http dependency.~

**Comment by sagebind** [comment]
This is no longer blocked, as [http 0.2](https://github.com/hyperium/http/releases/tag/v0.2.0) was released today!

**Comment by sagebind** [comment]
This change is now available in the [0.9](https://github.com/sagebind/isahc/releases/tag/0.9.0) release!
